pkg/virt-handler/virtwrap/api/schema.go

```diff
@@ -142,15 +142,31 @@ type DomainList struct {
 }
 
 type DomainSpec struct {
-	XMLName xml.Name `xml:"domain"`
-	Name    string   `xml:"name"`
-	UUID    string   `xml:"uuid,omitempty"`
-	Memory  Memory   `xml:"memory"`
-	Type    string   `xml:"type,attr"`
-	OS      OS       `xml:"os"`
-	SysInfo *SysInfo `xml:"sysinfo,omitempty"`
-	Devices Devices  `xml:"devices"`
-	Clock   *Clock   `xml:"clock,omitempty"`
+	XMLName  xml.Name     `xml:"domain"`
+	Type     string       `xml:"type,attr"`
+	XmlNS    string       `xml:"xmlns:qemu,attr,omitempty"`
+	Name     string       `xml:"name"`
+	UUID     string       `xml:"uuid,omitempty"`
+	Memory   Memory       `xml:"memory"`
+	OS       OS           `xml:"os"`
+	SysInfo  *SysInfo     `xml:"sysinfo,omitempty"`
+	Devices  Devices      `xml:"devices"`
+	Clock    *Clock       `xml:"clock,omitempty"`
+	Resource *Resource    `xml:"resource,omitempty"`
+	QEMUCmd  *Commandline `xml:"qemu:commandline,omitempty"`
+}
+
+type Commandline struct {
```

Comment:
We should switch to libvirt-xml-go quickly - instead of writing our own mapping.

Reply:
Or are there significant gaps?

Reply:
Constructing the structs like I like them, and then converting them to the XML expectations for libvirt works fine for me. Fields which are not there, are definitely fields we don't support, I like this kind of white-listing ...

Feel free to switch over to libvirt-xml-go, I don't see drawbacks, but I don't see much benefit either.

Reply:
The benefit is to avoid code duplication.

Reply:
Sure, I will not stop you doing the switch.